packages/bfCliTask/src/qnaMakerCommand.ts

```diff
@@ -0,0 +1,332 @@
+    private qnaTranslateKey: string;
+    private qnaSourceLang: string;
+    private qnaTargetLang: string;
+    private activeLearning: boolean;
+    private keyType: string;
+    private utils = new Utils();
+
+    constructor() {
+        this.qnaMakerSubCommand = taskLibrary.getInput('qnaMakerSubCommand', false) as string;
+        this.qnaKey = taskLibrary.getInput('qnaKey', false) as string;
+        this.kbName = taskLibrary.getInput('kbName', false) as string || "";
+        this.kbDTOFileLocation = this.utils.validatePath('kbDTOFileLocation');
+        this.publishNewKB = taskLibrary.getBoolInput('publishNewKB', false);
+        this.kbId = taskLibrary.getInput('kbId', false) as string;
+        this.feedbackRecordDTOLocation = taskLibrary.getPathInput('feedbackRecordDTOFileLocation',false) as string;
+        this.kbHostName = taskLibrary.getInput('kbHostName',false) as string;
+        this.kbEndPointKey = taskLibrary.getInput('kbEndPointKey',false) as string;
+        this.wordAlterationsDTOFileLocation = taskLibrary.getInput('wordAlterationsDTOFileLocation',false) as string;
+        this.serviceEndpoint = taskLibrary.getInput('serviceEndpoint',false) as string;
+        this.qnaQuestion = taskLibrary.getInput('qnaQuestion', false) as string;
+        this.isAlterationFile = taskLibrary.getBoolInput('isAlterationFile', false);
+        this.qnaConvertFilePathInput = taskLibrary.getPathInput('qnaConvertFilePathInput', false) as string;
+        this.qnaConvertFilePathOutput = taskLibrary.getPathInput('qnaConvertFilePathOutput', false) as string;
+        this.qnaTranslateInput = taskLibrary.getInput('qnaTranslateInput', false) as string;
+        this.qnaTranslateOutputFolder = taskLibrary.getInput('qnaTranslateOutputFolder', false) as string;
+        this.qnaTranslateKey = taskLibrary.getInput('qnaTranslateKey', false) as string;
+        this.qnaSourceLang = taskLibrary.getInput('qnaSourceLang', false) as string;
+        this.qnaTargetLang = taskLibrary.getInput('qnaTargetLang', false) as string;
+        this.activeLearning = taskLibrary.getBoolInput('activeLearning', false);
+        this.keyType = taskLibrary.getInput('keyType', false) as string;
+    }
+
+    public executeSubCommand = () => {
+        try {
+            switch (this.qnaMakerSubCommand) {
+                case 'CreateKB':
+                    this.createKnowledgeBase();
+                    break;
+                case 'DeleteKB':
+                    this.deleteKnowledgeBase();
+                    break;
+                case 'PublishKB':
+                    this.publishKnowledgeBase();
+                    break;
+                case 'ReplaceKB':
+                    this.replaceKnowledgeBase();
+                    break;
+                case 'UpdateKB':
+                    this.updateKnowledgeBase();
+                    break;
+                case 'TrainKB':
+                    this.traingKnowledgeBase();
```

Comment:
Here there's a typo. 
```suggestion
                    this.trainKnowledgeBase();
```